### PR TITLE
Remove unseen product-subscription tokens

### PR DESCRIPTION
**Pull request by rafax**
Replaces a long-lived TODO that caused disabled product-subscription tokens not to be cleared (so, licences with tokens disabled can still be used), [Slack](https://sourcegraph.slack.com/archives/C05497E9MDW/p1708017558275189).

Use of Redis KEYS is problematic, but using a better alternative like SCAN would require us to refactor our Redis client and add:

- scan support (easy)
- context support (major refactor of all Redis usages or doubling the Redis surface area)
- error support (see above)

so, this PR improves CG correctness with a minimal/incremental change.

From quick testing, fetching 2000 keys takes <5ms, fetching 2000000 keys takes ~1s.

## Test plan

- all new code covered by unit tests
- tested locally

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
good work!

File: cmd/cody-gateway/internal/actor/productsubscription/productsubscription.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"time"
 
 	"github.com/Khan/genqlient/graphql"
+	"github.com/amit7itz/goset"
```
[the remark this answers is not in the document]
Comment (rafax) [reply cmd/cody-gateway/internal/actor/productsubscription/productsubscription.go:11]: Sure, I found this one when searching for set in `go.mod`, but I guess our own type wouldn't be there.

[the remark this answers is not in the document]
Comment (rafax) [reply cmd/cody-gateway/internal/actor/productsubscription/productsubscription.go]: Done

**Review by thenamankumar — APPROVED** [review]

**Comment by bobheadxi** [comment]
Thanks for fixing this long-standing TODO ❤️

**Pull request merged**